### ISSUE TITLE
Low accuracy for sst2 dataset using merged model provided by src/ties_minimal.ipynb

### ISSUE THREAD

**Issue by Ldoun** [prateeky2806/ties-merging#8]
Thank you for your excellent work. I'm currently trying to reproduce your results using smaller examples, and while I've seen decent performance on the MNLI and RTE datasets, I'm encountering an issue with the SST-2 dataset.

Specifically, when using the provided src/ties_minimal.ipynb example, the merged model achieves 0.0% accuracy on SST-2. It appears to be generating labels like "neutral" or "contradiction," which are typically used in the MNLI dataset, rather than the expected SST-2 labels.

Could you please advise on how I can improve the merged model's performance for SST-2? 

Thanks,

Doun Lee

```
import os
import torch
from datasets import load_dataset
from promptsource.templates import DatasetTemplates, Template
from transformers import AutoTokenizer, AutoModelForSeq2SeqLM

from utils import state_dict_to_vector, vector_to_state_dict, ties

if __name__ == '__main__':
    # Model Loading
    tokenizer = AutoTokenizer.from_pretrained("t5-base")
    
    merged_state_dict = torch.load('merged_state_dict_origin.pt')

    model = AutoModelForSeq2SeqLM.from_pretrained("t5-base")
    print(model.load_state_dict(merged_state_dict))
    model.cuda()

    print('merged model loaded')

    # RTE Dataset
    dataset_name, subset_name = "glue", "rte"
    dataset = load_dataset("nyu-mll/glue", subset_name, split="validation")

    prompt = Template(
        name='custom',
        jinja="rte sentence1:{{sentence1}}sentence 2:{{sentence2}}|||{{answer_choices[label]}}",
        reference='',
        answer_choices='entailment ||| not entailment',
    )

    cnt = 0
    for i in range(10):    
        result = prompt.apply(dataset[i])
        input_ids = tokenizer(
            result[0], return_tensors="pt"
        ).input_ids.cuda()

        output = model.generate(input_ids) #, decoder_input_ids=decoder_input_ids)
        print(result[0])
        pred = tokenizer.decode(output[0], skip_special_tokens=True)
        print(f'{pred} => {result[1]}')
        cnt += pred.strip() == result[1].strip()
    print(cnt/10)

    print('*'*80)
    cnt = 0
    model_rte = AutoModelForSeq2SeqLM.from_pretrained("PavanNeerudu/t5-base-finetuned-rte").cuda()
    for i in range(10):    
        result = prompt.apply(dataset[i])
        input_ids = tokenizer(
            result[0], return_tensors="pt"
        ).input_ids.cuda()

        output = model_rte.generate(input_ids) #, decoder_input_ids=decoder_input_ids)
        print(result[0])
        pred = tokenizer.decode(output[0], skip_special_tokens=True)
        print(f'{pred} => {result[1]}')
        cnt += pred.strip() == result[1].strip()
    print(cnt/10)
    print()

    # MNLI dataset
    dataset_name, subset_name = "glue", "mnli"
    dataset = load_dataset("nyu-mll/glue", subset_name, split="train")

    prompt = Template(
        name='custom',
        jinja="mnli premise::{{premise}}hypothesis:{{hypothesis}}|||{{answer_choices[label]}}",
        reference='',
        answer_choices='entailment ||| neutral ||| contradiction',
    )

    cnt = 0
    for i in range(10):    
        result = prompt.apply(dataset[i])
        input_ids = tokenizer(
            result[0], return_tensors="pt"
        ).input_ids.cuda()

        output = model.generate(input_ids) #, decoder_input_ids=decoder_input_ids)
        print(result[0])
        pred = tokenizer.decode(output[0], skip_special_tokens=True)
        print(f'{pred} => {result[1]}')
        cnt += pred.strip() == result[1].strip()
    print(cnt/10)

    print('*'*80)
    cnt = 0
    model_rte = AutoModelForSeq2SeqLM.from_pretrained("PavanNeerudu/t5-base-finetuned-mnli").cuda()
    for i in range(10):    
        result = prompt.apply(dataset[i])
        input_ids = tokenizer(
            result[0], return_tensors="pt"
        ).input_ids.cuda()

        output = model_rte.generate(input_ids) #, decoder_input_ids=decoder_input_ids)
        print(result[0])
        pred = tokenizer.decode(output[0], skip_special_tokens=True)
        print(f'{pred} => {result[1]}')
        cnt += pred.strip() == result[1].strip()
    print(cnt/10)
    print()

    # SST2 dataset
    dataset_name, subset_name = "glue", "sst2"
    dataset = load_dataset("nyu-mll/glue", subset_name, split="validation")

    prompt = Template(
        name='custom',
        jinja="sst2 sentence:{{sentence}}|||{{answer_choices[label]}}",
        reference='',
        answer_choices='negative ||| positive',
    )

    cnt = 0
    for i in range(10):    
        result = prompt.apply(dataset[i])
        input_ids = tokenizer(
            result[0], return_tensors="pt"
        ).input_ids.cuda()

        output = model.generate(input_ids) #, decoder_input_ids=decoder_input_ids)
        print(result[0])
        pred = tokenizer.decode(output[0], skip_special_tokens=True)
        print(f'{pred} => {result[1]}')
        cnt += pred.strip() == result[1].strip()
    print(cnt/10)

    print('*'*80)
    cnt = 0
    model_rte = AutoModelForSeq2SeqLM.from_pretrained("PavanNeerudu/t5-base-finetuned-sst2").cuda()
    for i in range(10):    
        result = prompt.apply(dataset[i])
        input_ids = tokenizer(
            result[0], return_tensors="pt"
        ).input_ids.cuda()

        output = model_rte.generate(input_ids) #, decoder_input_ids=decoder_input_ids)
        print(result[0])
        pred = tokenizer.decode(output[0], skip_special_tokens=True)
        print(f'{pred} => {result[1]}')
        cnt += pred.strip() == result[1].strip()
    print(cnt/10)
```
